src/Channels/Memory.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+namespace Channels
+{
+    public struct Memory<T>
```

Comment:
I think there's an argument to be made for `Memory<T>` to replace `Bytes` in the "slices" lib. They serve the same purpose, with the key difference being that `Bytes` only supports `<byte>`. Best to stabilize usage first, though.